assets/src/frontendIndex.js

```diff
@@ -0,0 +1,20 @@
+import { SpreadsheetFrontend } from './blocks/Spreadsheet/SpreadsheetFrontend';
+
+const PLANET4_BLOCK_PREFIX = 'wp-block-planet4-blocks-';
+const COMPONENTS = {
+	spreadsheet: SpreadsheetFrontend
+};
+
+const getFrontendBlockComponent = blockNode => COMPONENTS[
+	Array.from(blockNode.classList)
+		.find( className => className.includes( PLANET4_BLOCK_PREFIX ))
```

Comment:
Maybe we can create a dedicated `data-` attribute for this instead of a class? Something like `data-render="spreadsheet"` Then we could get the block name from the attribute's value, instead of having to search the classnames and remove the prefix. [The querySelector below](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/273/files?file-filters%5B%5D=.js&file-filters%5B%5D=.php&file-filters%5B%5D=.scss&hide-deleted-files=true#diff-e318476ddd94efc2f71618da8e052ddbR14) would then also be a bit simpler, just matching any elements that have the data attribute, regardless of its value.
```
document.querySelectorAll(`[data-render]`).forEach(...)
```

Reply:
That was the initial approach, but then I couldn't find a way to get the block name in the `save()` method, it receives the `attributes` and a set of `className`s which includes the block's class and any "block style" classes. But I guess I can just make the block name a constant in the block registration constructor and use it in `save()`, and make it a prop of the `FrontendNode`. I think I'll also include the `classNames` (which I'm not doing now) so we can use block styles in the future (or any other CSS stuff).